### Step 1: Import necessary libraries

In [25]:
import pandas as pd
import numpy as np

### Step 2: Loading merged dataset from notebook 3

In [41]:
df = pd.read_csv('merged_cleaned_dataset_nb3.csv')

In [42]:
df.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0  93705     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
1  93706     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2  93708     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3  93709     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
4  93711     1.0     0.0     1.0     0.0     0.0     0.0     0.0     0.0   

   DPQ090  ...      WTMEC2YR  SDMVPSU  SDMVSTRA  INDHHIN2  INDFMIN2  INDFMPIR  \
0     0.0  ...   8338.419786      2.0     145.0       3.0       3.0      0.82   
1     0.0  ...   8723.439814      2.0     134.0       NaN       NaN       NaN   
2     0.0  ...  14372.488765      2.0     138.0       6.0       6.0      1.63   
3     NaN  ...  12277.556662      1.0     136.0       2.0       2.0      0.41   
4     0.0  ...  12390.919724      2.0     134.0      15.0      15.0      5.00   

   phq9_items_answered  phq9_total_score  probable_depression  \
0                    9               0.0                    0   
1                    9               0.0                    0   
2                    9               0.0                    0   
3                    0               NaN                    0   
4                    9               2.0                    0   

   suicidal_ideation  
0                  0  
1                  0  
2                  0  
3                  0  
4                  0  

[5 rows x 60 columns]

In [34]:
print("Number of columns:", df.shape[1])
df.columns.tolist()

Number of columns: 60


['SEQN',
 'DPQ010',
 'DPQ020',
 'DPQ030',
 'DPQ040',
 'DPQ050',
 'DPQ060',
 'DPQ070',
 'DPQ080',
 'DPQ090',
 'DPQ100',
 'SDDSRVYR',
 'RIDSTATR',
 'RIAGENDR',
 'RIDAGEYR',
 'RIDAGEMN',
 'RIDRETH1',
 'RIDRETH3',
 'RIDEXMON',
 'RIDEXAGM',
 'DMQMILIZ',
 'DMQADFC',
 'DMDBORN4',
 'DMDCITZN',
 'DMDYRSUS',
 'DMDEDUC3',
 'DMDEDUC2',
 'DMDMARTL',
 'RIDEXPRG',
 'SIALANG',
 'SIAPROXY',
 'SIAINTRP',
 'FIALANG',
 'FIAPROXY',
 'FIAINTRP',
 'MIALANG',
 'MIAPROXY',
 'MIAINTRP',
 'AIALANGA',
 'DMDHHSIZ',
 'DMDFMSIZ',
 'DMDHHSZA',
 'DMDHHSZB',
 'DMDHHSZE',
 'DMDHRGND',
 'DMDHRAGZ',
 'DMDHREDZ',
 'DMDHRMAZ',
 'DMDHSEDZ',
 'WTINT2YR',
 'WTMEC2YR',
 'SDMVPSU',
 'SDMVSTRA',
 'INDHHIN2',
 'INDFMIN2',
 'INDFMPIR',
 'phq9_items_answered',
 'phq9_total_score',
 'probable_depression',
 'suicidal_ideation']

### Step 3: Renaming all Demo_J columns

In [38]:
rename_dict = {
    "SDDSRVYR": "SurveyCycle_SDDSRVYR",
    "RIDSTATR": "InterviewStatus_RIDSTATR",
    "RIAGENDR": "Sex_RIAGENDR",
    "RIDAGEYR": "AgeYears_RIDAGEYR",
    "RIDAGEMN": "AgeMonths_RIDAGEMN",
    "RIDRETH1": "RaceEthnicity1_RIDRETH1",
    "RIDRETH3": "RaceEthnicity3_RIDRETH3",
    "RIDEXMON": "ExamMonth_RIDEXMON",
    "RIDEXAGM": "AgeAtExamMonths_RIDEXAGM",
    "DMQMILIZ": "MilitaryService_DMQMILIZ",
    "DMQADFC": "ActiveDutyFlag_DMQADFC",
    "DMDBORN4": "BornInUS_DMDBORN4",
    "DMDCITZN": "CitizenshipStatus_DMDCITZN",
    "DMDYRSUS": "YearsInUS_DMDYRSUS",
    "DMDEDUC3": "EducationYouth_DMDEDUC3",
    "DMDEDUC2": "EducationAdult_DMDEDUC2",
    "DMDMARTL": "MaritalStatus_DMDMARTL",
    "RIDEXPRG": "PregnancyStatus_RIDEXPRG",
    "SIALANG": "InterviewLanguage_SIALANG",
    "SIAPROXY": "InterviewProxy_SIAPROXY",
    "SIAINTRP": "InterviewInterpreter_SIAINTRP",
    "FIALANG": "FamilyInterviewLanguage_FIALANG",
    "FIAPROXY": "FamilyInterviewProxy_FIAPROXY",
    "FIAINTRP": "FamilyInterviewInterpreter_FIAINTRP",
    "MIALANG": "MobileExamLanguage_MIALANG",
    "MIAPROXY": "MobileExamProxy_MIAPROXY",
    "MIAINTRP": "MobileExamInterpreter_MIAINTRP",
    "AIALANGA": "AudioLanguage_AIALANGA",
    "DMDHHSIZ": "HouseholdSize_DMDHHSIZ",
    "DMDFMSIZ": "FamilySize_DMDFMSIZ",
    "DMDHHSZA": "HH_ChildrenUnder6_DMDHHSZA",
    "DMDHHSZB": "HH_Children6to17_DMDHHSZB",
    "DMDHHSZE": "HH_Adults60Plus_DMDHHSZE",
    "DMDHRGND": "HH_ReferenceGender_DMDHRGND",
    "DMDHRAGZ": "HH_ReferenceAgeGroup_DMDHRAGZ",
    "DMDHREDZ": "HH_ReferenceEducation_DMDHREDZ",
    "DMDHRMAZ": "HH_ReferenceMarital_DMDHRMAZ",
    "DMDHSEDZ": "HH_ReferenceSpouseEducation_DMDHSEDZ",
    "WTINT2YR": "InterviewWeight_WTINT2YR",
    "WTMEC2YR": "ExamWeight_WTMEC2YR",
    "SDMVPSU": "PSU_SDMVPSU",
    "SDMVSTRA": "Strata_SDMVSTRA",
    "INDHHIN2": "HouseholdIncome_INDDHHIN2",
    "INDFMIN2": "FamilyIncome_INDFMIN2",
    "INDFMPIR": "PovertyIncomeRatio_INDFMPIR"
}


In [39]:
df = df.rename(columns=rename_dict)
df.columns

Index(['SEQN', 'DPQ010', 'DPQ020', 'DPQ030', 'DPQ040', 'DPQ050', 'DPQ060',
       'DPQ070', 'DPQ080', 'DPQ090', 'DPQ100', 'SurveyCycle_SDDSRVYR',
       'InterviewStatus_RIDSTATR', 'Sex_RIAGENDR', 'AgeYears_RIDAGEYR',
       'AgeMonths_RIDAGEMN', 'RaceEthnicity1_RIDRETH1',
       'RaceEthnicity3_RIDRETH3', 'ExamMonth_RIDEXMON',
       'AgeAtExamMonths_RIDEXAGM', 'MilitaryService_DMQMILIZ',
       'ActiveDutyFlag_DMQADFC', 'BornInUS_DMDBORN4',
       'CitizenshipStatus_DMDCITZN', 'YearsInUS_DMDYRSUS',
       'EducationYouth_DMDEDUC3', 'EducationAdult_DMDEDUC2',
       'MaritalStatus_DMDMARTL', 'PregnancyStatus_RIDEXPRG',
       'InterviewLanguage_SIALANG', 'InterviewProxy_SIAPROXY',
       'InterviewInterpreter_SIAINTRP', 'FamilyInterviewLanguage_FIALANG',
       'FamilyInterviewProxy_FIAPROXY', 'FamilyInterviewInterpreter_FIAINTRP',
       'MobileExamLanguage_MIALANG', 'MobileExamProxy_MIAPROXY',
       'MobileExamInterpreter_MIAINTRP', 'AudioLanguage_AIALANGA',
       'HouseholdSize_D

### Step 4: Missingness per variable (count + proportion)

In [40]:
missing_by_var = pd.DataFrame({
    "missing_count": df.isna().sum(),
    "missing_pct": df.isna().mean() * 100
}).sort_values("missing_pct", ascending=False)

missing_by_var

missing_count  missing_pct
AgeMonths_RIDAGEMN                             5533   100.000000
AgeAtExamMonths_RIDEXAGM                       5271    95.264775
EducationYouth_DMDEDUC3                        5265    95.156335
ActiveDutyFlag_DMQADFC                         5006    90.475330
PregnancyStatus_RIDEXPRG                       4476    80.896440
YearsInUS_DMDYRSUS                             3839    69.383698
HH_ReferenceSpouseEducation_DMDHSEDZ           2780    50.243991
DPQ100                                         2174    39.291524
AudioLanguage_AIALANGA                         1377    24.887041
PovertyIncomeRatio_INDFMPIR                     734    13.265859
DPQ060                                          450     8.133020
DPQ080                                          449     8.114947
DPQ090                                          448     8.096873
DPQ040                                          448     8.096873
DPQ030                                          447     8.078800
DPQ070                                          447     8.078800
DPQ010                                          447     8.078800
DPQ050                                          446     8.060727
DPQ020                                          446     8.060727
phq9_total_score                                440     7.952286
MobileExamLanguage_MIALANG                      389     7.030544
MobileExamProxy_MIAPROXY                        389     7.030544
MobileExamInterpreter_MIAINTRP                  389     7.030544
HouseholdIncome_INDDHHIN2                       282     5.096693
MaritalStatus_DMDMARTL                          268     4.843665
EducationAdult_DMDEDUC2                         268     4.843665
FamilyInterviewInterpreter_FIAINTRP             266     4.807519
FamilyInterviewLanguage_FIALANG                 266     4.807519
FamilyInterviewProxy_FIAPROXY                   266     4.807519
FamilyIncome_INDFMIN2                           266     4.807519
HH_ReferenceEducation_DMDHREDZ                  251     4.536418
HH_ReferenceMarital_DMDHRMAZ                    122     2.204952
CitizenshipStatus_DMDCITZN                        3     0.054220
BornInUS_DMDBORN4                                 0     0.000000
MilitaryService_DMQMILIZ                          0     0.000000
ExamMonth_RIDEXMON                                0     0.000000
SEQN                                              0     0.000000
SurveyCycle_SDDSRVYR                              0     0.000000
Sex_RIAGENDR                                      0     0.000000
InterviewStatus_RIDSTATR                          0     0.000000
AgeYears_RIDAGEYR                                 0     0.000000
RaceEthnicity1_RIDRETH1                           0     0.000000
RaceEthnicity3_RIDRETH3                           0     0.000000
InterviewProxy_SIAPROXY                           0     0.000000
HH_Adults60Plus_DMDHHSZE                          0     0.000000
HH_Children6to17_DMDHHSZB                         0     0.000000
HH_ChildrenUnder6_DMDHHSZA                        0     0.000000
FamilySize_DMDFMSIZ                               0     0.000000
HouseholdSize_DMDHHSIZ                            0     0.000000
InterviewInterpreter_SIAINTRP                     0     0.000000
InterviewLanguage_SIALANG                         0     0.000000
HH_ReferenceAgeGroup_DMDHRAGZ                     0     0.000000
PSU_SDMVPSU                                       0     0.000000
ExamWeight_WTMEC2YR                               0     0.000000
InterviewWeight_WTINT2YR                          0     0.000000
HH_ReferenceGender_DMDHRGND                       0     0.000000
Strata_SDMVSTRA                                   0     0.000000
phq9_items_answered                               0     0.000000
probable_depression                               0     0.000000
suicidal_ideation                                 0     0.000000

**Comments:** This table summarizes the number and percentage of missing values for each variable in the merged dataset, sorted from highest to lowest missingness. Several demographic variables (e.g., AgeMonths_RIDAGEMN, AgeAtExamMonths_RIDEXAGM, EducationYouth_DMDEDUC3) show very high missingness because they only apply to specific subgroups (e.g., infants or youth). The PHQ-9 items (DPQ010–DPQ100) each have about ~8% missingness, which is relatively consistent across questions. Key identifiers and core demographics (e.g., SEQN, Sex_RIAGENDR, AgeYears_RIDAGEYR, RaceEthnicity variables) have no missing values. This pattern suggests that most missingness is structural (not applicable) rather than due to data corruption, but variables with very high missingness may need to be excluded or handled carefully in later analyses.

### Step 5: Missingness per participant

In [44]:
df["missing_per_participant"] = df.isna().sum(axis=1)
df[["SEQN", "missing_per_participant"]].head()

SEQN  missing_per_participant
0  93705                        8
1  93706                       13
2  93708                        6
3  93709                       22
4  93711                        5

** Comments: ** This step calculates how many variables are missing for each participant in the merged dataset. The missing_per_participant column represents the total number of missing values across all variables for that individual. For example, SEQN 93706 has 13 missing values, while SEQN 93708 has 6 missing values. This helps identify participants with a large amount of missing data who may need to be excluded from certain analyses. Examining missingness at the participant level is important to ensure that analytic results are not driven by individuals with very incomplete records.

### Step 6: Checking if missingness is related to depression status

In [45]:
df.groupby("probable_depression")["missing_per_participant"].agg(["count", "mean", "median", "max"])

count      mean  median  max
probable_depression                              
0                     5072  8.116916     7.0   29
1                      461  6.895879     7.0   14

**Comments:** (Missingness vs. Depression Status):
This table summarizes the amount of missing data per participant, grouped by depression status. Participants without probable depression (0) have an average of about 8.12 missing variables, while participants with probable depression (1) have a slightly lower average of about 6.90 missing variables. The median number of missing variables is the same for both groups (7), suggesting similar typical levels of missingness. The maximum number of missing variables is higher among non-depressed participants (29) compared to depressed participants (14). Overall, missingness does not appear to be substantially higher among depressed individuals, indicating that excluding participants with missing data is unlikely to disproportionately remove depressed participants from the analytic sample.